File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int p1 = 0;
        int p2 = n-1;
        while((numbers[p1] + numbers[p2]) != target){
            if((numbers[p1] + numbers[p2]) < target){
                p1++;
            } else{
                p2--;
            }
        }
        return {p1+1,p2+1};
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
from collections import defaultdict

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = defaultdict(int)
        for i, num in enumerate(nums):
            d[num] = i

        for i, num in enumerate(nums):
            conj = target-num
            if conj in d and d[conj] != i:
                return [i, d[conj]]
        
